### DISCUSSION TITLE
Creating Adapter/API for Orbit

### DISCUSSION THREAD

**Discussion by edwinnglabs** [alan-turing-institute/sktime#1389] [Ideas]
Hi all,

I am exploring chance to incorporate an API for [orbit](https://github.com/uber/orbit).  I wonder how I could start the proposal and if approved, how I could proceed.  For example, should I consider building a similar API like those mentioned [here](https://www.sktime.org/en/latest/examples/04_benchmarking.html)?

**Comment by fkiraly** [comment]
That's a great idea!

The challenge with `orbit` is that the few algorithms in it are probabilistic forecasters, and `sktime` does not have a fully consolidated API for probabilistic forecasting yet, see the discussion in this issue https://github.com/alan-turing-institute/sktime/issues/984
But it's definitely on the roadmap - my idea/preference would be to use `tensorflow-probability` probability distributions as the return in `predict_proba` - these can easily be vectorized. For variational posteriors, these would be parametric distributions, for MCMC posteriors, these would be empirical distributions.

The "right way" would be first writing a short STEP how the probabilistic forecasting interface (`predict_proba` etc) would look like - that would be in this repository https://github.com/sktime/enhancement-proposals.
Before that (or, if detailed enough, instead of it) you could also reply to #984; at least, you should say you are interested in working on this and/or what you are going to do.

The STEP would then be reviewed by core developers, and once it's accepted as a way forward, you would make a PR to `sktime` which is then reviewed etc as usual. Unusually, it would probably change the forecaster base class (`BaseForecaster`), extension template, and/or other central API classes, which is why this is a little more complex than just adding an estimator.

Does this make sense? Let me know of any questions.
FYI @aiwalter, @mloning.

**Comment by fkiraly** [comment]
Of course, you could also go for a "simpler" interface which only produces interval forecasts (that are currently already part of the API), that would require no STEP and only an "extension" PR.

**Reply by fkiraly** [reply]
for this, you would take the predictive distributions that `orbit` produces, and in the interface compute the requested quantiles. For `predict`, you would take the posterior predictive mean/expectation.
